From c194a9f6fba702ced2e7b5771a792a517bbed033 Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <leio@gentoo.org>
Date: Wed, 16 Jul 2014 16:31:27 +0300
Subject: [PATCH 3/3] gl: Include GL_CFLAGS while checking for GL typedefs

Otherwise those checks may fail at configure time if they contain extra
include paths, while at build time they are included, potentially causing
incompatible typedefs between system GL headers and gstreamer compatibility
prototypes.
---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index 8261ee2..e87081c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1343,6 +1343,9 @@ GST_GL_HAVE_GLCHAR=0
 GST_GL_HAVE_GLSIZEIPTR=0
 GST_GL_HAVE_GLINTPTR=0
 
+old_CFLAGS=$CFLAGS
+CFLAGS="$GL_CFLAGS $CFLAGS"
+
 AC_CHECK_TYPES(GLeglImageOES, [], [], [[$GL_INCLUDES]])
 if test "x$ac_cv_type_GLeglImageOES" = "xyes"; then
   GST_GL_HAVE_GLEGLIMAGEOES=1
@@ -1363,6 +1366,8 @@ if test "x$ac_cv_type_GLintptr" = "xyes"; then
   GST_GL_HAVE_GLINTPTR=1
 fi
 
+CFLAGS=$old_CFLAGS
+
 GL_CONFIG_DEFINES="$GL_CONFIG_DEFINES
 #define GST_GL_HAVE_GLEGLIMAGEOES $GST_GL_HAVE_GLEGLIMAGEOES
 #define GST_GL_HAVE_GLCHAR $GST_GL_HAVE_GLCHAR
-- 
1.9.1

